subprojects/build-init/src/main/groovy/org/gradle/buildinit/plugins/internal/maven/Maven2Gradle.groovy

```diff
@@ -374,30 +475,70 @@ version = '$project.version'""";
     def packSources = { sourceSets ->
         def sourceSetStr = ''
         if (!sourceSets.empty) {
-            sourceSetStr = """task packageSources(type: Jar) {
+            switch (scriptDsl) {
+                case KOTLIN:
+                    sourceSetStr = """val packageSources by tasks.creating(Jar::class) {
```

Comment:
Is there a reified version that can be used here?

Reply:
Nope, see above